include/storage/io_config.hpp

```diff
@@ -0,0 +1,82 @@
+#ifndef IO_CONFIG_HPP
+#define IO_CONFIG_HPP
+
+#include <boost/filesystem/path.hpp>
+
+namespace osrm
+{
+namespace storage
+{
+struct IOConfig
+{
+    IOConfig() = default;
```

Comment:
Yes, I would prefer to drop this as a constructor, but in practice it doesn't work: https://github.com/Project-OSRM/osrm-backend/blob/bf6698f4cc7608a75d9ac72f18248356b6ed162c/src/tools/extract.cpp#L27 is how command line arguments are parsed, they require an instantiated version of the Config files. I started without that and without the `UseDefaultOutputNames` function, but I couldn't get it to work like that.